src/Client.php

```diff
@@ -257,63 +268,60 @@ public function getGuzzleClient() {
    * @param array $body Request body to be converted to JSON.
    * @param array $query Query params
    *
-   * @return array Returns the status code and json_decoded body contents.
+   * @return array|bool Returns json_decoded body contents or FALSE.
    */
   protected function request($method, $uri, array $body = [], array $query = []) {
     $requestOptions = [];
 
     if ($method != 'DELETE') {
       $requestOptions = [
-        'query' => is_array($query) ? $query : [],
-        'body' => is_array($body) ? json_encode($body) : $body,
+        'query' => $query,
+        'body' => json_encode($body),
       ];
     }
 
     try {
       $response = $this->guzzleClient->request($method, $uri, $requestOptions);
-      $code = $response->getStatusCode();
-      $body = $response->getBody()->getContents();
     } catch (RequestException $exception) {
-      $code = $exception->getCode();
-      $body = $exception->getResponse()->getBody()->getContents();
+      // @todo Handle errors.
+      return FALSE;
     }
 
-    return [
-      'response' => $code,
-      'body' => json_decode($body)
-    ];
+    return json_decode($response->getBody()->getContents(), true);
```

Comment:
This change means that only the response json (as an array) is returned, instead of the code/response tuple. This makes reactive error handling basically impossible, but makes for insanely clean and abstracted code.

For example `getBuildConfig('wcms-d8-shepherd')` returns either the build config array, or false, instead of an array containing both the http response and build config / error message. This means you can do things like:

```
if (!getBuildConfig('foo')) {
    createBuildConfig('foo', 'bar', ...);
}
```

Swift solves this problem with Optionals.